### PR TITLE
Compatibility with Semigroup/Monoid proposal

### PR DESCRIPTION
**Pull request by bgamari**

### CODE REVIEW

**Comment by kazu-yamamoto** [comment]
I confirmed that this PR works well.

@vincenthz Please merge this and release new versions of x509 and x509-store.

**Comment by vincenthz** [comment]
@kazu-yamamoto I'm getting to this slowly (my bottom to top approach doesn't scale very well with included yak shaking). I'm not pulling anything until the CI is correctly integrated

File: x509-store/Data/X509/CertificateStore.hs [hunk 1]
```diff
@@ -26,12 +31,22 @@ import qualified Data.ByteString as B
 data CertificateStore = CertificateStore (M.Map DistinguishedName SignedCertificate)
                       | CertificateStores [CertificateStore]
 
+#if MIN_VERSION_base(4,9,0)
+instance Semigroup CertificateStore where
+    (<>) = append
```
Comment (ggreif) [inline x509-store/Data/X509/CertificateStore.hs]: why not
``` haskell
    (<>) = mappend
```
Then you won't need the outlining if `append`...

Comment (vincenthz) [reply x509-store/Data/X509/CertificateStore.hs]: as I understand mappend is disappearing from Monoid in the future, and being replaced by a simple `mappend = <>`

Comment (ggreif) [reply x509-store/Data/X509/CertificateStore.hs]: I see. Thanks for the info!

**Pull request merged**